dht/i_partitioner.hh

```diff
@@ -648,6 +650,47 @@ future<utils::chunked_vector<partition_range>> split_range_to_single_shard(const
 
 std::unique_ptr<dht::i_partitioner> make_partitioner(sstring name);
 
+// Filter owned tokens using token_range_vector.
+// Conforms to FlattenedConsumerFilterV2 so it can be used
+// with flat_mutation_reader_v2::consume_in_thread.
+class tokens_filter {
+    token_range_vector _sorted_owned_ranges;
```

Comment:
incremental_token_filter ?

Reply:
I don't think this implementation detail must be reflected in the class name.
For example, it could in theory be constructed with a unordered set of arbitrary token_ranges that it could sort and deoverlap to achieve the required invariant.

Reply:
But the incremental nature is reflected in its interface as a requirement for correct usage, not only in the implementation. E.g. incremental reader selector is named so because you expect the reads to be in incremental token order.

Reply:
Fixed in v2 by refactoring incremental_owned_ranges_checker out of cleanup compaction